### PR TITLE
added extracting xml values into json object example

### PR DESCRIPTION
**Pull request by fortunkam**
Show extracting specific values from an xml document into a json object variable and how to use that object in policy expressions and liquid templates.

### CODE REVIEW

**Comment by mikebudzynski** [comment]
@fortunkam, thanks for the contribution. Could you please include the contributed policy in the TOC? https://github.com/Azure/api-management-policy-snippets/blob/master/examples/README.md

**Comment by simonkurtz-MSFT** [comment]
Please rename _extracting multiple values from xml documents.xml_ to _Extracting multiple values from xml documents.policy.xml_

File: examples/extracting multiple values from xml documents.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            var doc = XDocument.Parse(((IResponse)context.Variables["sendRequestResponse"]).Body.As<string>());
+
+            var obj = new JObject();
+            obj["conditionValue"] = doc.Descendants().Where(r=>r.Name.LocalName == "conditionValue").First().Value;
+            obj["value1"] = doc.Descendants().Where(r=>r.Name.LocalName == "value1").First().Value;
+            obj["value3"] = doc.Descendants().Where(r=>r.Name.LocalName == "value3").First().Value;
+
+            return obj;
+
+            }" />
```
Comment (simonkurtz-MSFT) [inline examples/extracting multiple values from xml documents.xml:40]: Fix indentation

**Review by simonkurtz-MSFT — COMMENTED** [review]
Looks like this will all work. Please consider trimming some of the whitespace lines a bit to condense the code.

**Comment by fortunkam** [comment]
> Looks like this will all work. Please consider trimming some of the whitespace lines a bit to condense the code.

I was torn on the whitespace because it made the example more readable.

File: examples/Extracting multiple values from xml documents.policy.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<!-- The policy defined in this file demonstrates how to work with multiple extract values from an xml document. -->
+<!-- If you wanted to assign each of the values from a document to a variable you need to parse the document multiple times which is inefficient.
+     The alternative is to parse the document once and extract the values into a json object variable.  The individual json properties can then be used in both policy expressions and liquid templates.   
+-->
+<!-- The following example assumes a send request call is made in the inbound policy that returns an xml document into a variable.
```
Comment (simonkurtz-MSFT) [inline examples/Extracting multiple values from xml documents.policy.xml:5]: Needs a `-->` at the end of line 5.

**Comment by simonkurtz-MSFT** [comment]
@mikebudzynski, I verified the mechanics of this by creating a temporary endpoint that returns an XML response to then be ingested by this code. Other than my outstanding suggestions, I think this is good.

**Comment by mikebudzynski** [comment]
> @mikebudzynski, I verified the mechanics of this by creating a temporary endpoint that returns an XML response to then be ingested by this code. Other than my outstanding suggestions, I think this is good.

Thanks, Simon! I will wait for Matthew to address the outstanding comments.

File: examples/Extracting multiple values from xml documents.policy.xml (hunk 2 above)
Comment (fortunkam) [reply examples/Extracting multiple values from xml documents.policy.xml:5]: oops, will fix now

**Comment by fortunkam** [comment]
@mikebudzynski Should be all complete now, just pending the final review

File: examples/Extracting multiple values from xml documents.policy.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                <!-- Do something when condition is met -->
+                <trace source="My API" severity="verbose">
+                    <message>Condition was met</message>
+                </trace>
+            </when>
+        </choose>
+
+        <!-- Or in liquid transformations like this.. -->
+
+        <set-body template="liquid">
```
[the remark this answers is not in the document]
Comment (simonkurtz-MSFT) [reply examples/Extracting multiple values from xml documents.policy.xml:63]: Hi @fortunkam, what do you think about this suggestion?

[the remark this answers is not in the document]
Comment (fortunkam) [reply examples/Extracting multiple values from xml documents.policy.xml:63]: @simonkurtz-MSFT I did wonder about this but thought it detracted from the core message (i.e. this is how you use it in liquid templates).  Am happy to add it in, if you think it makes the example better?

**Comment by simonkurtz-MSFT** [comment]
Hi @fortunkam, I would include it because it paints a more complete picture for the reader, especially if the reader is a bit more on the novice side. The higher-fidelity these snippets are the better, in my opinion.

**Comment by fortunkam** [comment]
updated @simonkurtz-MSFT

**Comment by simonkurtz-MSFT** [comment]
LGTM! Thank you!

**Comment by mikebudzynski** [comment]
Thanks for the contribution!

**Review by mikebudzynski — APPROVED** [review]

**Pull request merged**